windows file: URLs

Looks like Windows compatibility got broken with borgstore 0.1.0 (or recent borg changes):

```
self = <[AttributeError("'Store' object has no attribute 'levels'") raised in repr()] Store object at 0x2072b4d2a10>
url = 'file://D:/a/_temp/msys64/tmp/pytest-of-runneradmin/pytest-0/popen-gw1/test_create_topical_archiver_0/repository'
backend = None
levels = {'archives/': [0], 'cache/': [0], 'config/': [0], 'data/': [0], ...}

    def __init__(self, url: Optional[str] = None, backend: Optional[BackendBase] = None, levels: Optional[dict] = None):
        self.url = url
        if backend is None and url is not None:
            backend = get_backend(url)
            if backend is None:
>               raise BackendURLInvalid(f"Invalid Backend Storage URL: {url}")
E               borgstore.backends.errors.BackendURLInvalid: Invalid Backend Storage URL: file://D:/a/_temp/msys64/tmp/pytest-of-runneradmin/pytest-0/popen-gw1/test_create_topical_archiver_0/repository
```

Also, there seems to be an issue with `Store.levels` initialization, see first line.

--- a/src/borgstore/backends/posixfs.py
+++ b/src/borgstore/backends/posixfs.py
@@ -4,6 +4,7 @@ Filesystem based backend implementation - uses files in directories below a base
 
 import os
 import re
+import sys
 from pathlib import Path
 import shutil
 import stat
@@ -25,10 +26,20 @@ def get_file_backend(url):
     # - the third slash is NOT optional, it is the start of an absolute path as well
     #   as the separator between the host and the path part.
     # - the caller is responsible to give an absolute path.
-    file_regex = r"""
-        file://  # only empty host part is supported
-        (?P<path>(/.*))  # path must be an absolute path
+    # - windows: see there: https://en.wikipedia.org/wiki/File_URI_scheme
+    windows_file_regex = r"""
+        file://  # only empty host part is supported.
+        /  # 3rd slash is separator ONLY, not part of the path.
+        (?P<drive_and_path>([a-zA-Z]:/.*))  # path must be an absolute path.
     """
+    file_regex = r"""
+        file://  # only empty host part is supported.
+        (?P<path>(/.*))  # path must be an absolute path. 3rd slash is separator AND part of the path.
+    """
+    if sys.platform in ("win32", "msys", "cygwin"):
+        m = re.match(windows_file_regex, url, re.VERBOSE)
+        if m:
+            return PosixFS(path=m["drive_and_path"])
     m = re.match(file_regex, url, re.VERBOSE)
     if m:
         return PosixFS(path=m["path"])

